### ISSUE TITLE
Response.clone does not clone in-progress streaming body

### ISSUE THREAD

**Issue by robere2** [oven-sh/bun#6348]
### What version of Bun is running?

1.0.4

### What platform is your computer?

Linux 5.10.60.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

[Calling the `clone()` method on a `Response` object should return an identical `Response` object, including the body](https://developer.mozilla.org/en-US/docs/Web/API/Response/clone). Instead, it returns a `Response` that's missing the body, but otherwise identical in every other way (that I've found).

Minimal reproducible example:
```js
const res = await fetch("https://github.com/");
console.log("Clone Response length:", (await res.clone().text()).length)
console.log("Main Response length:", (await res.text()).length)
```

### What is the expected behavior?

The program should output two lines to the console, and each should contain the same integer number.

### What do you see instead?

The first line will contain the number "0" while the second line contains the expected value.

**Comment by Hanaasagi** [comment]
Related to https://github.com/oven-sh/bun/issues/4765

**Comment by Electroid** [comment]
Duplicate of #4765

**Comment by dangeredwolf** [comment]
Regression of #913, related to #6969. Also seems to be a duplicate of #1381 and #2477.  This is a pretty big issue breaking a lot of things, I hope someone will finally pay attention to it because of how many things this bug effects. There's already an open PR for it, #1693, oh, and in #6468 as well!

**Comment by liz3** [comment]
Theres a open pr for ages but the maintainers don't prio such fixes over other things:
https://github.com/oven-sh/bun/pull/6468

**Comment by BoltDoggy** [comment]
Maybe Polyfill

```
Request.prototype.clone = function (this: Request) {
  const { body } = this;
  const teedX = body?.tee();
  Object.defineProperty(this, "body", {
    get() {
      return teedX?.[0] || null;
    },
  });  return new Request(this, {
    body: teedX?.[1] || null,
  });
};
```

More Idea

```
Request.prototype.breakClone = function (this: Request) {
  const {
    url,
    method,
    headers,
    cache,
    credentials,
    destination,
    integrity,
    keepalive,
    mode,
    redirect,
    referrer,
    referrerPolicy,
    signal,
    bodyUsed,
    body,
    clone,
  } = this;
  const teedX = body?.tee();
  const arrayBufferPromise = new Response(teedX?.[0]).arrayBuffer();
  const teedY = teedX?.[1].tee();
  Object.defineProperty(this, "body", {
    get() {
      return teedY?.[0] || null;
    },
  });
  return {
    url,
    method,
    headers,
    cache,
    credentials,
    destination,
    integrity,
    keepalive,
    mode,
    redirect,
    referrer,
    referrerPolicy,
    signal,
    bodyUsed,
    body: teedY?.[1] || null,
    clone,
    arrayBuffer: async () => arrayBufferPromise,
    json: async () => new Response(await arrayBufferPromise).json(),
    text: async () => new Response(await arrayBufferPromise).text(),
    blob: async () => new Response(await arrayBufferPromise).blob(),
    formData: async () => new Response(await arrayBufferPromise).formData(),
  };
};

```

**Comment by vitch** [comment]
I've just run into this...

I was trying to show the `text()` of a `Response` if `response.json()` failed.

This gives an error:

```ts
try {
  let responseData = await response.json();
} catch (e) {
  let text = await response.text();
  // ...
}
```

> Body already used

According to [this SO answer](https://stackoverflow.com/a/54115314) using `response.clone()` is the correct way around this. But in bun `1.1.20` it's failing for me and the `clone`ed `body()` is an empty string...

**Comment by bholmesdev** [comment]
We've run into this issue when implementing our Actions pattern in Astro. We rely on `clone()` to internally handle incoming requests, and allow the user to read that body from their request handler as well. This bug is unfortunately a blocker for those using the latest version of Astro with Bun.

I'd love to see a fix implemented for this!

**Comment by mangs** [comment]
@bholmesdev This is the PR adding this functionality, I'm blocked on this too. https://github.com/oven-sh/bun/pull/12473

**Comment by jadbox** [comment]
This is a blocker for us as we also need to clone the request for our tooling, including for @bholmesdev Astro Actions.

**Comment by jadbox** [comment]
I'm not sure if this helps, but I have a small update from Jarred on a Bun/Astro workaround until request cloning is fixed.

![Screenshot_20240818-090649.png](https://github.com/user-attachments/assets/c611cbea-554a-459a-a648-30703f054fe9)

**Comment by QuentinDutot** [comment]
Got the same error while trying to use bun with h3 (v2)
```
await request.json() // works
await request.clone().json() // doesn't work
await new Request(request).json() // doesn't work
```

**Comment by Jarred-Sumner** [comment]
The fix for this will be included in Bun v1.1.27
